ports/psoc6/machine_pdm_pcm.c

```diff
@@ -511,21 +463,7 @@ static void mp_machine_pdm_pcm_init_helper(machine_pdm_pcm_obj_t *self, mp_arg_v
     self->sample_rate = args[ARG_sample_rate].u_int;
     self->decimation_rate = args[ARG_decimation_rate].u_int;
 
-    // Set clock values
```

Comment:
I am missing a check where we raise an exception if the clock is not properly configured for the desired rate?

Reply:
Sorry it was a miss when I wanted to bring clock related changes. It's back there now!